### DISCUSSION TITLE
it would be great if tree explorer can capture the mouse on opening

### DISCUSSION THREAD

**Discussion by gene-liang** [nvim-tree/nvim-tree.lua#2596] [Q&A]
I have keys mapped to mouse wheel scroll down/up, so when I hit a key to open the tree explorer, then hit those keys to let wheel scroll, it may not work the tree (depending on if the tree is the last window have the mouse captured), it's  inconvenient for me, because I have to grab the mouse to move to the tree and scroll, after that then I can have the keys work on the tree. It would be great if the tree can capture the mouse when user opens it.
I believe this feature is reasonable and useful because when we open the tree, we intend to work on the tree.

**Comment by geril07** [comment]
I think it's how neovim working, scroll executes on window under cursor? Is it even possible to implement with current nvim design? Did you try to use `<C-d>,<C-u>,<C-b>,<C-f>` for scrolling?

**Comment by gene-liang** [comment]
> I think it's how neovim working, scroll executes on window under cursor? Is it even possible to implement with current nvim design? Did you try to use `<C-d>,<C-u>,<C-b>,<C-f>` for scrolling?

Thanks for the comments, those keys you mentioned are about cursors, those I am referring are the mouse behaviors, namely the <scrollwheelup/down/left/right>, and it's the mouse lost focus when tree opened, not cursor, so those keys you mentioned work fine, but not for the scrollwheel (mapped) keys.

**Comment by gegoune** [comment]
Transferring to discussion. If you believe this is nvim-tree issue please open new issue following issue template.

**Comment by mohamedarish** [comment]
> it's inconvenient for me, because I have to grab the mouse to move to the tree and scroll, after that then I can have the keys work on the tree. It would be great if the tree can capture the mouse when user opens it.

Do you mean focus on the position of the cursor rather than the position of the mouse pointer?

The part being scrolled is the one in which the cursor is currently in.
If cursor is in tree, then scroll tree. If it is in the buffer, scroll the buffer?

The current behavior is to check the position of the mouse pointer.
I don't think the cursor ever loses focus if it is inside the window of the terminal, at least not on macOS 14.3
